test/js-api/memory/constructor.any.js

```diff
@@ -158,6 +201,30 @@ test(() => {
   assert_Memory(memory, { "size": 1, "index": "i64" });
 }, "Memory with i64 index constructor");
 
+test(() => {
+  const argument = { "initial": "3" };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 3 });
+}, "Memory with string value for initial");
+
+test(() => {
+  const argument = { "index": "i64", "initial": "3" };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 3 });
+}, "Memory with string value for initial (i64)");
+
+test(() => {
+  const argument = { "initial": true };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1 });
+}, "Memory with boolean value for initial");
+
+test(() => {
+  const argument = { "index": "i64", "initial": true };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1 });
+}, "Memory with boolean value for initial (i64)");
```

Comment:
I find it kind of that true can coerce silently to 1n, but 1 itself (i.e. the number) does not.   I guess that is the strange thing about bigint in JS.. seems out of place.